tests/test_manubot.py

```diff
@@ -67,3 +70,36 @@ def test_read_jsons():
     assert 'violet' in user_variables['namespace_1']['rainbow']
     assert 'yellow' in user_variables['namespace_2']['rainbow']
     assert 'orange' in user_variables['namespace_3']['rainbow']
+
+
+def test_add_author_affiliations_empty():
+    variables = {}
+    variables['authors'] = [
+        {'name': 'Jane Roe'},
+        {'name': 'John Doe'},
+    ]
+    returned_variables = add_author_affiliations(variables)
+    assert variables is returned_variables
+    assert 'affiliations' not in variables
+    for author in variables['authors']:
+        assert 'affiliation_numbers' not in author
+
+
+def test_add_author_affiliations():
+    variables = {}
+    variables['authors'] = [
+        # Legacy affiliations format (as a string that's `: ` separated)
+        {'name': 'Jane Roe', 'affiliations': 'Department of Doe, University of Roe; Peppertea University'},
+        # Prefered affiliations format as a list
+        {'name': 'John Doe', 'affiliations': ['Unique University', 'Peppertea University']},
+    ]
+    returned_variables = add_author_affiliations(variables)
+    assert variables is returned_variables
+    assert variables['affiliations'] == [
+        {'affiliation': 'Department of Doe, University of Roe', 'affiliation_number': 1},
+        {'affiliation': 'Peppertea University', 'affiliation_number': 2},
+        {'affiliation': 'Unique University', 'affiliation_number': 3},
+    ]
+    authors = variables['authors']
+    assert authors[0]['affiliation_numbers'] == [1, 2]
+    assert authors[1]['affiliation_numbers'] == [3, 2]
```

Comment:
Currently the [build fails](https://travis-ci.org/greenelab/manubot/builds/326477396#L1250) because of an ordering issue in pandas. From https://github.com/pandas-dev/pandas/issues/18776#issuecomment-356053890:

![pandas-inner-merge-order](https://user-images.githubusercontent.com/1117703/34686273-0d62e90e-f479-11e7-8491-fe3d0ae6d8c5.png)

When I wrote the code, I intended for the order of the numbers to match the order author affiliations were listed in `metadata.yaml`. However, one could also argue that numbers should always be sorted.

In other words, sorted numbers would force:

> Daniel Himmelstein<sup>1,2</sup>

Is that good or bad. Should author superscripts match the order they input in `metadata.yaml`. If yes, we could have:

> Daniel Himmelstein<sup>2,1</sup>

@agitter what do you think the ideal behavior is here?